### DISCUSSION TITLE
Error found

### DISCUSSION THREAD

**Discussion by mistersaputra** [ibnux/Android-SMS-Gateway#10] [Q&A] [answered]
Halo kak, terimakasih sourcodenya.
Saya ingin menggunakan program ini, tapi saya menemukan beberapa error. 
Apakah kakak bisa membantu untuk mencari solusinya?
Berikut error yang saya temukan:

**ObjectBox.java**
pada file ini ada import file MyObjectBox namun file tersebut tidak ada.
`import com.ibnux.smsgateway.data.MyObjectBox;`

**LogAdapter.java**
pada file ini AndroidStudio tidak bisa mengenali LogLine_
```
if (search.length() > 0) {
            datas = ObjectBox.get().boxFor(LogLine.class).query().contains(LogLine_.message, search).orderDesc(LogLine_.time).build().find(offset, limit);
        } else {
            datas = ObjectBox.get().boxFor(LogLine.class).query().orderDesc(LogLine_.time).build().find(offset, limit);
        }
```

**Comment by ibnux** [comment, marked as answer]
Pelajari ObjectBox disini
https://docs.objectbox.io/getting-started

ObjectBox harus sukses build dulu, baru object dengan underscore akan terbuat (LogLine_)

kadang bagian ini dikomentarin dulu, lalu build project, setelah sukses, hapus komentarnya

https://github.com/ibnux/Android-SMS-Gateway/blob/a742d6484504d134e81bccf0ba527f7654c7c191/app/src/main/java/com/ibnux/smsgateway/ObjectBox.java#L11-L13

**Reply by mistersaputra** [reply]
Setelah komen bagian ini dan masukin google-service.json yang didownload dari firebase terus build project jadi bisa kak. Makasih sarannya.
